internal/internal_task_handlers.go

```diff
@@ -846,11 +860,12 @@ func (w *workflowExecutionContextImpl) ProcessWorkflowTask(workflowTask *workflo
 	metricsTimer := metricsHandler.Timer(metrics.WorkflowTaskReplayLatency)
 
 	eventHandler.ResetLAWFTAttemptCounts()
+	eventHandler.sdkFlags.resetSdkFlags()
```

Comment:
I am not sure you want to reset SDK flags every task. I think the idea behind many flags (including this one, though maybe not all in the future) is that they only need to be set on any workflow task to apply to all future workflow tasks.

Reply:
>is that they only need to be set on any workflow task to apply to all future workflow tasks.
Yes  I agree, I'll rename this function since it's confusing